File: 0001-0500/0006/0006.zigzag-conversion.py
#
# @lc app=leetcode id=6 lang=python3
#
# [6] Zigzag Conversion
#

# @lc code=start
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1:
            return s
        res = [''] * numRows
        row = 0
        step = 1
        for c in s:
            res[row] += c
            if row == 0:
                step = 1
            elif row == numRows - 1:
                step = -1
            row += step
        return ''.join(res) 
# @lc code=end

File: 0001-0500/0006/0006.zigzag-conversion.js
/*
 * @lc app=leetcode id=6 lang=javascript
 *
 * [6] Zigzag Conversion
 */

// @lc code=start
/**
 * @param {string} s
 * @param {number} numRows
 * @return {string}
 */
var convert = function (s, numRows) {
    if (numRows == 1) return s;
    let len = s.length;
    let res = '';
    for (let i = 0; i < numRows; i++) {
        let j = i;
        while (j < len) {
            res += s[j];
            if (i != 0 && i != numRows - 1) {
                let k = j + 2 * (numRows - i - 1);
                if (k < len) {
                    res += s[k];
                }
            }
            j += 2 * numRows - 2;
        }
    }
    return res;
};
// @lc code=end

